### PR TITLE
Debug menu improvements + Teleport to maps

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #1813 

Also adds some usability improvements to remember indices.

This code is nasty and due for a refactor. I would advise just testing it instead of trying to review some of these diffs...

### CODE REVIEW

**Comment by Ghabry** [comment]
> This code is nasty and due for a refactor.

Haha. I wanted to add something to that scene one week ago and thought the same. The code is really hard to follow/extend. :/

**Comment by Ghabry** [comment]
guess this doesn't work in emscripten? In the web player I only get a blackscreen while testing. ( https://easyrpg.org/play/pr1817/?test-play )

Trying it again locally later.

**Comment by fmatthew5876** [comment]
Gaps are supported. You'll see an empty entry in the list and it'll do nothing when you select it.

Area maps appear in the list but are disabled

**Comment by fmatthew5876** [comment]
Added support for Full Heal to the Debug Menu.

I want to add call common event, but that's going to be in 6.2 as it comes after #1779 .

This PR is finished.

**Comment by fmatthew5876** [comment]
Regarding this black screen on emscripten, I see this too.

Maybe it's an async bug related to enabling teleports from scenes other than the map? Do we have any test game setup with quick access to an Escape or Teleport spell?

I suspect this might be a more general issue and if so, they would exhibit the same problem.

**Comment by fmatthew5876** [comment]
emscripten issue is fixed for debug menu and escape/teleport skills.

**Comment by fmatthew5876** [comment]
In chat @fdelapena commented that this PR doesn't handle map bounds. Unfortunately I don't think that can be done easily as we actually have to load the lmu file to know the bounds, which would incredibly slow on normal platforms and basically impossible on emscripten.

It's a debug feature, so I guess not a huge problem for now.

**Comment by Ghabry** [comment]
Minor suggestion: When Full Heal is invoked play SFX_UseItem

**Comment by fmatthew5876** [comment]
> 
> 
> Minor suggestion: When Full Heal is invoked play SFX_UseItem

Good idea, added this

File: src/scene_map.cpp [hunk 1]
```diff
@@ -50,6 +50,7 @@ Scene_Map::~Scene_Map() {
 }
 
 void Scene_Map::Start() {
+	Scene_Debug::ResetPrevIndicies();
```
Comment (Ghabry) [inline src/scene_map.cpp]: typo: Indices

Comment (fmatthew5876) [reply src/scene_map.cpp]: Fixed thanks
